src/test/java/com/artipie/settings/RepoDataTest.java

```diff
@@ -0,0 +1,234 @@
+import com.artipie.asto.fs.FileStorage;
+import com.artipie.asto.memory.InMemoryStorage;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.stream.Collectors;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+/**
+ * Test for {@link RepoData}.
+ * @since 0.1
+ */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+class RepoDataTest {
+
+    /**
+     * Test repository name.
+     */
+    private static final String REPO = "my-repo";
+
+    /**
+     * Temp dir.
+     * @checkstyle VisibilityModifierCheck (500 lines)
+     */
+    @TempDir
+    Path temp;
+
+    /**
+     * Test settings storage.
+     */
+    private BlockingStorage stngs;
+
+    /**
+     * Test settings storage.
+     */
+    private Storage storage;
+
+    /**
+     * Test data storage.
+     */
+    private BlockingStorage data;
+
+    @BeforeEach
+    void init() {
+        this.storage = new InMemoryStorage();
+        this.stngs = new BlockingStorage(this.storage);
+        this.data = new BlockingStorage(new FileStorage(this.temp));
+    }
+
+    @Test
+    void removesData() {
+        this.stngs.save(
+            new Key.From(String.format("%s.yml", RepoDataTest.REPO)),
+            this.repoSettings().getBytes(StandardCharsets.UTF_8)
+        );
+        this.data.save(new Key.From(RepoDataTest.REPO, "first.txt"), new byte[]{});
+        this.data.save(new Key.From(RepoDataTest.REPO, "second.txt"), new byte[]{});
+        new RepoData(this.storage)
+            .remove(new RepositoryName.Flat(RepoDataTest.REPO)).toCompletableFuture().join();
+        MatcherAssert.assertThat(
+            "Repository data are removed",
+            this.data.list(Key.ROOT).isEmpty()
+        );
+    }
+
+    @Test
+    void movesData() {
+        this.stngs.save(
+            new Key.From(String.format("%s.yml", RepoDataTest.REPO)),
+            this.repoSettings().getBytes(StandardCharsets.UTF_8)
+        );
+        this.data.save(new Key.From(RepoDataTest.REPO, "first.txt"), new byte[]{});
+        this.data.save(new Key.From(RepoDataTest.REPO, "second.txt"), new byte[]{});
+        new RepoData(this.storage)
+            .move(new RepositoryName.Flat(RepoDataTest.REPO), new RepositoryName.Flat("new-repo"))
+            .toCompletableFuture().join();
+        MatcherAssert.assertThat(
+            "Repository data are moved",
+            this.data.list(Key.ROOT).stream().map(Key::string)
+                .collect(Collectors.toList()),
+            Matchers.contains("new-repo/first.txt", "new-repo/second.txt")
+        );
+    }
+
+    @ParameterizedTest
+    @ValueSource(
+        strings = {"_storage.yaml", "_storage.yml", "my-repo/_storage.yaml", "my-repo/_storage.yml"}
```

Comment:
There was a mistake in front: settings file is actually called `_storageS.yaml` (check `com.artipie.settings.StorageAliases#FILE_NAME` constant). Please, correct the test

Reply:
Thanks for important notes
I have added according fix, @olenagerasimova @dgarus could you please to revew again?

Reply:
Thanks for important notes
I have added according fix, @olenagerasimova @dgarus could you please to revew again?